astroplan/scheduling.py

```diff
@@ -350,22 +559,21 @@ def from_timespan(cls, center_time, duration, **kwargs):
         duration : `~astropy.units.Quantity` or `~astropy.time.TimeDelta`
             Duration of time-span to schedule
         """
-        start_time = center_time - duration/2.
-        end_time = center_time + duration/2.
+        start_time = center_time - duration / 2.
+        end_time = center_time + duration / 2.
         return cls(start_time, end_time, **kwargs)
 
     def _make_schedule(self, blocks):
-
         # Combine individual constraints with global constraints, and
         # retrieve priorities from each block to define scheduling order
         _all_times = []
         _block_priorities = np.zeros(len(blocks))
-        for i,b in enumerate(blocks):
+        for i, b in enumerate(blocks):
             if b.constraints is None:
                 b._all_constraints = self.constraints
             else:
                 b._all_constraints = self.constraints + b.constraints
-            b._duration_offsets = u.Quantity([0*u.second, b.duration/2, b.duration])
+            b._duration_offsets = u.Quantity([0 * u.second, b.duration / 2, b.duration])
```

Comment:
I've been thinking about this over the weekend and I'm more convinced that an extra keyword on ```Constraints``` is the way to do this. To illustrate my thinking, I'll set out my case below, and then provide some code which sets out how I think it might work. Feel free to ignore this comment if you disagree with the approach.

Firstly, the time spacing on which a ```Constraint``` needs to be calculated is intrinsically a property of the ```Constraint``` itself. Some ```Constraints``` change rapidly and need to be evaluated on a fine grid; others change slowly and can be evaluated on a coarse grid, or just at the start of the ```ObservingBlock```. So a ```time_resolution``` keyword should be a property of the ```Constraint```. 

Second, I'm not sure that this conflicts with the way ```Constraint```s are defined. A call to a constraint is still evaluated at a single point, so the current code and concept remain unchanged. 

All that changes is how the ```Scheduler```s evaluate the constraints over an ```ObservingBlock```, and I don't think this has to be more complex. This is probably best illustrated by some example code for the SequentialScheduler:

```python
constraint_res = []
for constraint in b._all_constraints:
    times = current_time + transition_time + constraint.grid(b.duration)
    constraint_res.append(constraint(self.observer, [b.target], times)
# things are a bit trickier here, because constraint_res is not square array
# i.e rows are different shapes
block_constraint_results.append(
    np.prod([np.prod(res) for res in constraint_res])
)
```

The ```constraint.grid``` call would create an array of ```TimeDelta```s with the appropriate spacing if ```b.duration``` was greater than ```constraint.time_resolution``` or zero if not (or if ```constraint.time_resolution``` was negative) 

This would probably make things more efficient; for example and ```AirmassConstraint``` would be evaluate a few times over a block, whereas we only really need to evaluate a Lunar constraint at the start time, for example. It would also fix my issues with the ```MeridianTransitConstraint```, which would get a negative ```time_resolution``` keyword.